Popup dialog

DrabWeb opened the pull request:
Also changes `PauseButton` to a generic `DialogButton`

DrabWeb committed:
Added popup dialog

DrabWeb committed:
Speedup triangles

DrabWeb committed:
Even faster

DrabWeb committed:
Merge https://github.com/ppy/osu into popup-dialog

DrabWeb committed:
New design

DrabWeb committed:
Merge https://github.com/ppy/osu into popup-dialog

DrabWeb committed:
Update to work with latest framework

DrabWeb committed:
Removed background triangles

DrabWeb committed:
Conform to naming conventions

DrabWeb committed:
Remove SpaceTextOnHover

peppy commented on osu.Game/Graphics/UserInterface/DialogButton.cs:
```diff
@@ -13,30 +13,41 @@
-        private Color4 buttonColour;
-        public Color4 ButtonColour
+        private Color4 colour;
+        public new Color4 Colour
```
i think we're avoiding `new` overrides like this at the moment (still not 100% sure it's the best way) so ButtonColour is probably best here.

DrabWeb committed:
Remove new override in DialogButton

DrabWeb committed:
Enter key presses first OK button, 1-9 presses button at number

DrabWeb committed:
Forgot returns

DrabWeb committed:
Remove uneccesary cast, move enter OK button to one liner

DrabWeb committed:
Added beatmap delete dialog

peppy commented on osu.Game/Screens/Select/BeatmapDeleteDialog.cs:
```diff
@@ -0,0 +1,53 @@
+            }
+        }
+
+        public BeatmapDeleteDialog()
```
why not just accept the beatmap in the ctor? it should be readonly anyway

DrabWeb committed:
Added DialogManager(not wired up for desktop project yet)

DrabWeb committed:
DialogManager -> DialogOverlay, cleaned up how BeatmapDeletDialog works, added global DialogOverlay

DrabWeb committed:
Testing build

DrabWeb committed:
Made DialogOverlay use PopupDialog's direct valfixed last dialogs not expiring, DialogOverlay actually hides now, fixed an issue where multiple buttons could be pressed at once

peppy committed:
Merge branch 'master' into popup-dialog

peppy committed:
Post-merge fixes.

peppy committed:
Tidy up code.

The triggered bool is not even necessary because input is no longer handled after the OverlayContainer's state is set to hidden.

peppy committed:
More code tidying.

peppy committed:
Fix incorrect event order.

peppy commented:
👍
